src/Filter/Strings.php

```diff
@@ -39,6 +39,9 @@ public static function filter(
             return null;
         }
 
+        self::checkIfScalarAndConvert($value);
+        self::checkIfObjectAndConvert($value);
+        self::validateIfObjectIsAString($value);
```

Comment:
I suggest doing something like this
```php
try {                                                                                                                   
    $stringValue = (function (string $str) : string {                                                                   
        return $str;                                                                                                    
    })($value);                                                                                                         
} catch (TypeError $e) {                                                                                                
    throw new FilterException("Value '" . var_export($value, true) . "' is not a string");                                                                        
}  
```

It will allow strings, scalars and objects with __tostring